### ISSUE TITLE
About performance

### ISSUE THREAD

**Issue by crapthings** [cult-of-coders/grapher#266]
METEOR@1.6.1.1
cultofcoders:grapher@1.3.6

i've tested my db and got performance issue.

```js
GroupMembers.addLinks({
  group: {
    type: 'one',
    collection: Groups,
    field: 'groupId',
  },
  user: {
    type: 'one',
    collection: Users,
    field: 'userId',
  },
})

const groupmembers = GroupMembers.createQuery({
  user: {
    username: 1,
  },
  group: {
    name: 1,
  },
}).fetch()
```

I20180625-10:04:58.430(8)?   ... 6583 more items ]
I20180625-10:04:58.430(8)? undefined: 1285.079ms

this is another test by fetch all content and assemble manually

```js
const users = Users.find().fetch()
const groups = Groups.find().fetch()

const usersKeyById = _.keyBy(users, '_id')
const groupsKeyById = _.keyBy(groups, '_id')

const groupmembers = GroupMembers.find().map(gm => ({
  user: usersKeyById[gm.userId],
  groups: groupsKeyById[gm.groupId],
  ...gm,
}))
```

I20180625-10:05:58.910(8)?   ... 6583 more items ]
I20180625-10:05:58.910(8)? undefined: 285.375ms

**Comment by crapthings** [comment]
it looks like query was transform into mongo aggregate

https://github.com/cult-of-coders/grapher/issues/188#issuecomment-359643600

**Comment by theodorDiaconu** [comment]
@crapthings give me a quick reproduction repo so I can play with this. It may be related to $lookup indexes as you said.

**Comment by theodorDiaconu** [comment]
Reproduced this. With Grapher 470ms, without Grapher: 70ms.
I will find out the culprit! This got me really curious :D

**Comment by theodorDiaconu** [comment]
I expect for that kind of scenario Grapher to be a little bit slower, because it has overhead, but not 5x slower..

**Comment by theodorDiaconu** [comment]
Found the culprit. I have managed to make Grapher 5x faster. This is mind-blowing..

**Comment by theodorDiaconu** [comment]
@crapthings this has been fixed in 1.3.7. You will do it faster if you do it manually just because you don't apply filters and the database returns your responses much faster, so it's not necessarily a Grapher disadvantage. Anyway, I tested for the following:

1000 users
1000 groups
Each user has 2 groups, so 2 group memberships per user ~2000

Results using your way: 58ms
Results using Grapher: 70ms

This is acceptable if we take into account that it retrieves only the Users and Groups that you need. (In our case all bc the use-case is built that way, but you get it!)

**Comment by crapthings** [comment]
@theodorDiaconu great! will test
